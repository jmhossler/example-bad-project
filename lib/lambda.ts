import * as cdk from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda';
import * as pythonLambda from '@aws-cdk/aws-lambda-python';

export class Example extends cdk.Construct {
    public handler: lambda.Function;

    constructor(scope: cdk.Construct, id: string) {
        super(scope, id);

        this.handler = new pythonLambda.PythonFunction(this, `${id}`, {
            functionName: `${id}`,
            timeout: cdk.Duration.seconds(10),
            entry: "resources/",
            index: "example.py",
            handler: "hello_world",
            runtime: lambda.Runtime.PYTHON_3_8
        });
    }
}
