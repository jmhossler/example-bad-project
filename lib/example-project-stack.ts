import * as cdk from '@aws-cdk/core';
import { Example } from "./lambda";

export class ExampleProjectStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const lambda = new Example(this, `${id}-lambda`);
  }
}
