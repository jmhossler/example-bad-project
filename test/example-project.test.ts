import { expect as expectCDK, matchTemplate, MatchStyle, SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as ExampleProject from '../lib/example-project-stack';

describe('Example Stack', () => {
    const app = new cdk.App();
    const stack = new ExampleProject.ExampleProjectStack(app, 'MyTestStack');

    test("Snapshot", () => {
        expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });
});
