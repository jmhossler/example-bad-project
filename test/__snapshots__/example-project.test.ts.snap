// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Stack Snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfArtifactHashD450F8A3": Object {
      "Description": "Artifact hash for asset \\"a05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decf\\"",
      "Type": "String",
    },
    "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfS3Bucket93EF3424": Object {
      "Description": "S3 bucket for asset \\"a05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decf\\"",
      "Type": "String",
    },
    "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfS3VersionKeyFF5EA998": Object {
      "Description": "S3 key for asset version \\"a05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decf\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "MyTestStacklambda247C431A": Object {
      "DependsOn": Array [
        "MyTestStacklambdaServiceRole1F504B92",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfS3Bucket93EF3424",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfS3VersionKeyFF5EA998",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersa05f7567dc05667291b6a5b07b2874c6bedbfdf490a0becb62478340f773decfS3VersionKeyFF5EA998",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "FunctionName": "MyTestStack-lambda",
        "Handler": "example.hello_world",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyTestStacklambdaServiceRole1F504B92",
            "Arn",
          ],
        },
        "Runtime": "python3.8",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyTestStacklambdaServiceRole1F504B92": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
